### ISSUE TITLE
Optimize  annotations.go `deleteLabels()` 

### ISSUE THREAD

**Issue by djwhatle** [konveyor/mig-controller#608]
**Summary**

- `deleteLabels()` iterates every possible GVR for in every namespace in migration and runs a discovery client `list()` followed by discovery client `update()` on all `list()` results.
- `deleteLabels()` will (presumably) start completely over in the next reconcile if it encounters a conflict error. If there are enough resources, the possibility of one resource having a conflictError is high, meaning this _could stall forever._
- `deleteLabels()` (aka `EnsureLabelsDeleted`) has been observed taking hours to complete on complex migrations. 

[Link to relevant code](https://github.com/konveyor/mig-controller/blob/master/pkg/controller/migmigration/annotation.go#L450)

**Solutions to consider**

- Do we need to remove labels on every possible GVK? Can we just run removal over a subset of GVKs that we actually label?
- Can we add label retries for `conflictErrors`?  This may allow us to get through this operation on the first try
- Can we drop this function altogether? Does it matter if we leave labels around?

```
// deleteLabels will delete all migration.openshift.io/migrated-by labels
// from the application upon successful completion
func (t *Task) deleteLabels() error {
	client, GVRs, err := gvk.GetGVRsForCluster(t.PlanResources.DestMigCluster, t.Client)
	if err != nil {
		return liberr.Wrap(err)
	}

	listOptions := k8sclient.MatchingLabels(map[string]string{
		MigratedByLabel: string(t.Owner.UID),
	}).AsListOptions()

	for _, gvr := range GVRs {
		for _, ns := range t.destinationNamespaces() {
			list, err := client.Resource(gvr).Namespace(ns).List(*listOptions)
			if err != nil {
				return liberr.Wrap(err)
			}
			for _, r := range list.Items {
				labels := r.GetLabels()
				delete(labels, MigratedByLabel)
				r.SetLabels(labels)
				_, err = client.Resource(gvr).Namespace(ns).Update(&r, metav1.UpdateOptions{})
				if err != nil {
					// The part touches the application on the destination side, fail safe to ensure
					// this won't block the migration
					if k8serror.IsMethodNotSupported(err) {
						continue
					}
					return liberr.Wrap(err)
				}
			}
		}
	}

	return nil
}
```

**Comment by djwhatle** [comment]
Would like to find the concrete list of resources that we attach `migration.openshift.io/migrated-by` so that we can cut down on the number of unlabeling requests we send. Seems like every GVK may be  excessive.

**Comment by djwhatle** [comment]
Doing some initial profiling to understand how much time this step takes. Will write back here.

**Comment by djwhatle** [comment]
More context:

- Label is added to every resource we restore in our plugin, merged in openshift-migration-plugin#58 
- Label is used in "deletedMigrated" to do a Migration Failure Rollback (which is disabled by default now)
